elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -527,7 +520,7 @@ public void removeRelation(String fieldName, PersistentResource removeResource)
      * @param newRelation the new relation
      */
     public void addRelation(String fieldName, PersistentResource newRelation) {
-        checkFieldAwarePermissions(UpdatePermission.class, fieldName);
+        checkFieldAwarePermissions(UpdatePermission.class, fieldName, newRelation, null);
```

Comment:
The existing value may have been a collection and not null.  Again we need to look at the type of teh relationship.